### PR TITLE
dev-lang/python: use --with-build-python for cross builds for 3.11

### PR DESCRIPTION
**Pull request by thesamesam**
Tested with sparc64-unknown-linux-gnu.

Closes: https://bugs.gentoo.org/847910
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

File: dev-lang/python/python-3.11.0_beta3.ebuild [hunk 1]
```diff
@@ -203,6 +204,44 @@ src_configure() {
...
+		mkdir "${WORKDIR}"/${P}-${CBUILD} || die
+		pushd "${WORKDIR}"/${P}-${CBUILD} &> /dev/null || die
+		ECONF_SOURCE="${S}" econf_build "${myeconfargs_cbuild[@]}"
+
+		# Avoid as many dependencies as possible for the cross build.
+		local mod
+		for mod in NIS _DBM _GDBM _DBM _SQLITE3 _HASHLIB _SSL _CURSES \
+			_CURSES_PANEL READLINE _TKINTER PYEXPAT ; do
+			echo "MODULE_${mod}=disabled" >> Makefile || die
+		done
```
Comment (mgorny) [inline dev-lang/python/python-3.11.0_beta3.ebuild:235]: Perhaps just inline the full list with `<<-EOF`?

**Comment by chewi** [comment]
It's not as bad as I expected, but I wish we could do it a different way. Are there really no changes needed to BDEPEND? Not even zlib or something?

File: dev-lang/python/python-3.11.0_beta3.ebuild (hunk 1 above)
Comment (thesamesam) [reply dev-lang/python/python-3.11.0_beta3.ebuild:235]: Sure.

**Comment by thesamesam** [comment]
> It's not as bad as I expected, but I wish we could do it a different way. Are there really no changes needed to BDEPEND? Not even zlib or something?

Dropped zlib too. Built python and libpython have no external deps.

File: dev-lang/python/python-3.11.0_beta3.ebuild [hunk 2]
```diff
@@ -203,6 +204,54 @@ src_configure() {
...
+		MODULE__DBM=disabled
+		MODULE__SQLITE3=disabled
+		MODULE__HASHLIB=disabled
+		MODULE__SSL=disabled
+		MODULE__CURSES=disabled
+		MODULE__CURSES_PANEL=disabled
+		MODULE_READLINE=disabled
+		MODULE__TKINTER=disabled
+		MODULE_PYEXPAT=disabled
+		MODULE_ZLIB=disabled
```
Comment (mgorny) [inline dev-lang/python/python-3.11.0_beta3.ebuild:244]: Indent, plz.

**Pull request merged**